fetch/nosniff/parsing-nosniff.window.js

```diff
@@ -0,0 +1,21 @@
+promise_test(() => fetch("resources/x-content-type-options.json").then(res => res.json()).then(runTests), "Loading JSON…");
+
+function runTests(allTestData) {
+  for (let i = 0; i < allTestData.length; i++) {
+    const testData = allTestData[i],
+          input = encodeURIComponent(testData.input);
+    async_test(t => {
+      const script = document.createElement("script");
+      t.add_cleanup(() => script.remove());
+      if (testData.nosniff) {
```

Comment:
I live completely in network land, so am far from knowledgeable about Javascript or the web platform, beyond HTTP (Admittedly, in Chrome, that also includes the MIME sniffing logic), but looking at the updated test, I still find it non-obvious why the nosniff results in errors.  The sniffed as text -> treated as valid Javascipt connection isn't obvious to me.

That having been said, I do think that this makes it much clearer that the onerror cases correspond to nosniff being respected.

Maybe just something along the lines of "The resource must be sniffed as test to be successfully loaded as a script."